superbit_lensing/medsmaker/superbit/medsmaker_mocks.py

```diff
@@ -411,97 +454,208 @@ def make_psf_models(self, select_stars=False, use_coadd=False):
-            truthcat = 'truth_gaussJitter_001.dat'
+        # If flagged, get a "clean" star catalog for PSFEx input
+        if select_truth_stars==True:
+            # This will break for any truth file nomenclature that isn't pipeline default
+            import glob
+            truthdir=self.data_dir
+            truthcat = glob.glob(''.join([truthdir,'*truth.fits']))[0]
             truthfilen=os.path.join(truthdir,truthcat)
             self.logprint("using truth catalog %s" % truthfilen)
-            psfcat_name = self._select_stars_for_psf(sscat=imcat_ldac_name,truthfile=truthfilen)
+            psfcat_name = self._select_stars_for_psf(sscat=im_cat,truthfile=truthfilen)
 
         else:
-            psfcat_name=imcat_ldac_name
+            psfcat_name = im_cat
 
         # Now run PSFEx on that image and accompanying catalog
         psfex_config_arg = '-c '+sextractor_config_path+'psfex.mock.config'
-        # Will need to make that tmp/psfex_output generalizable
         outcat_name = imagefile.replace('.fits','.psfex.star')
         cmd = ' '.join(['psfex', psfcat_name,psfex_config_arg,'-OUTCAT_NAME',
                             outcat_name])
         self.logprint("psfex cmd is " + cmd)
         os.system(cmd)
         # utils.run_command(cmd, logprint=self.logprint)
-        """
-        psfex_name_tmp1=(imcat_ldac_name.replace('.ldac','.psf'))
-        psfex_name_tmp2= psfex_name_tmp1.split('/')[-1]
-        psfex_model_file='/'.join([self.psf_path,psfex_name_tmp2])
-        """
+
         psfex_model_file=imcat_ldac_name.replace('.ldac','.psf')
+
         # Just return name, the make_psf_models method reads it in as a PSFEx object
         return psfex_model_file
 
-    def _select_stars_for_psf(self,sscat,truthfile):
+    def _make_piff_model(self, imagefile, weightfile='weight.fits', sextractor_config_path=None,
+            psfex_out_dir='./tmp/', select_truth_stars=False,star_params=None):
+        '''
+        Method to invoke PIFF for PSF modeling
+        Also creates individual exposure catalogs
+        Unclear what it should return yet...
+        Steps:
+            - Create sextractor catalog for input image
+            - Filter sexcat to create star catalog on the fly
+              with some common-sense selections
+            - If option is given, match star cat against truth
+            - Run PIFF
+            -
+
+        First, let's get it to run on one, then we can focus on running list!
+        '''
+
+        if sextractor_config_path is None:
+                sextractor_config_path = os.path.join(self.base_dir, 'superbit/astro_config/')
+
+        imcat_ldac_name=imagefile.replace('.fits','_cat.ldac')
+
+        if select_truth_stars==True:
+            # This will break for any truth file nomenclature
+            # that isn't pipeline default
+            import glob
+            truthdir=self.data_dir
+            try:
+                truthcat = glob.glob(os.path.join(truthdir,'truth*.dat'))[0]
```

Comment:
the default is now to output a `fits` table - I think it makes more sense to try that _first_

Reply:
Sure; the `.dat` is for backwards compatibility anyway.

Reply:
Done (I hope!)